pkg/extra/handlers.go

```diff
@@ -63,13 +110,66 @@ func (a *API) handleConsent(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func NewAPI(service ServiceInterface, kratos kratos.ServiceInterface, logger logging.LoggerInterface) *API {
+func (a *API) shouldEnforceWebAuthnWithSession(ctx context.Context, session *client.Session) (bool, error) {
+	// enforce only if one of the authentication methods was oidc
+	for _, method := range session.AuthenticationMethods {
+		if method.Method != nil && *method.Method == "oidc" {
+			webAuthnAvailable, err := a.kratos.HasWebAuthnAvailable(ctx, session.Identity.GetId())
+			if err != nil {
+				return false, err
+			}
+			return !webAuthnAvailable, nil
+		}
+	}
+	return false, nil
+}
+
+func (a *API) webAuthnSettingsRedirect(w http.ResponseWriter, returnTo string) error {
+	redirect, err := url.JoinPath("/", a.contextPath, "/ui/setup_passkey")
+	if err != nil {
+		return err
+	}
+
+	errorId := "session_aal2_required"
+
+	// Set the original consent URL as return_to, to continue the flow after WebAuthn key is set
+	redirectTo, err := url.ParseRequestURI(redirect)
+	if err != nil {
+		return err
+	}
+
+	q := redirectTo.Query()
+	q.Set("return_to", returnTo)
+	redirectTo.RawQuery = q.Encode()
+	redirectPath := redirectTo.String()
+
+	w.WriteHeader(http.StatusSeeOther)
+	_ = json.NewEncoder(w).Encode(
+		kratos.ErrorBrowserLocationChangeRequired{
+			Error:             &client.GenericError{Id: &errorId},
+			RedirectBrowserTo: &redirectPath,
+		},
+	)
+	return nil
+}
+
+func NewAPI(service ServiceInterface, kratos kratos.ServiceInterface, baseURL string, oidcWebAuthnSequencingEnabled bool, logger logging.LoggerInterface) *API {
 	a := new(API)
 
 	a.service = service
 	a.kratos = kratos
 
 	a.logger = logger
 
+	a.baseURL = baseURL
+	a.oidcWebAuthnSequencingEnabled = oidcWebAuthnSequencingEnabled
+
+	fullBaseURL, err := url.Parse(baseURL)
+	if err != nil {
+		// this should never happen if app is configured properly
+		a.logger.Fatalf("Failed to construct API base URL: %v\n", err)
+	}
+	a.contextPath = fullBaseURL.Path
```

Comment:
We pass the `baseURL` and convert it to `contextPath`. We store both variables in the struct, but only contextPath is used. I think we should keep only one of the 2.

Since baseURL is used in the kratos module as well, I would suggest to keep only that to stay consistent

Reply:
Both `baseURL` and `contextPath` are stored in the struct in [kratos](https://github.com/canonical/identity-platform-login-ui/blob/8acccc89e48df4946921f518a182c546220a8a74/pkg/kratos/handlers.go#L28-L29C2) module, so I think it's already consistent